lib/ffmpeg/transcoderbase.py

```diff
@@ -205,7 +205,12 @@ def transcode(self):
           "{}_{}_{}.yuv".format(self.case, n, channel))
         osyuv = filepath2os(yuv)
         vppscale = self.get_vpp_scale(self.width, self.height, "sw")
+        codec = output["codec"]
+        mode = output["mode"]
+        encoder = self.get_encoder(codec, mode)
         iopts = "-i {}"
+        if ("av1_qsv") == encoder:
+          iopts = "-c:v av1_qsv -i {}" 
```

Comment:
This needs to be more general so that it can work with other hwaccel plugins (qsv, vaapi, etc).  Basically, we'll need to setup the hwaccel like we do in `gen_input_opts`.  Also, this is the only special case at the moment, so it should be something like:
```python
iopts = ""
if output["codec"] in ["av1"]:
  # WA: no SW decoder for AV1 in ffmpeg, currently
  ###
  # TODO: setup hwaccel stuff
  ###
  decoder = self.get_decoder(output["codec"], "hw")
  iopts += f"-c:v {decoder}"

iopts += " -i {}"
```

Reply:
Option 2: It might be possible to use the new Decoder classes in lib/ffmpeg/qsv/decoder.py, lib/ffmpeg/vaapi/decoder.py, etc. with some minor changes.

Reply:
thanks for you advice, ffmpeg-qsv is passed.But i need to modify /lib/ffmpeg/vaapi/transcoder.py av1 codec when  running ffmpeg-vaapi. If not the {decoder} is av1_qsv,It is still not supported during verification.

Reply:
I'd prefer not to introduce too many special-case hacks.  Please create an internal JIRA and assign to me... this corner-case was not anticipated in the original code and may require more refactoring than expected.  I have some ideas and may be able to help by providing a solution or examples.

Reply:
OK, I created a interal JIRA and assign to you ,thanks for your help.